test/functorch/common_utils.py

```diff
@@ -268,18 +269,33 @@ def f(dummy, *args, **kwargs):
     outer_in_dims = (0,) + in_dims
     batched_args, kwarg_values = maybe_clone_inputs()
     output = vmap(vmap(f, inner_in_dims), outer_in_dims)(dummy, *batched_args, **kwarg_values)
-    yield (expected, output)
+
+    yield (loop_out, batched_out, expected, output)
```

Comment:
What is the benefit of yielding everything here over the previous approach? This leads to callsites of get_fallback_and_vmap_exhaustive needing to do two assertions instead of just one.

Reply:
Here's an idea for how to organize the code:
- we have a `_compute_quantities_for_vmap_test` that yields (loop_out, batched_out, expected, output)
- we have `get_fallback_and_vmap_exhaustive` call `_compute_quantities_for_vmap_test`, but it can do two yields instead of one:
```
for quantities in compute_quantities_for_vmap_test(
                op, batched_args, kwarg_values, in_dims, out_dim, batch_size,
                compute_loop_out=False):
            assert quantities[0] is None
            final = (expected_batched, ) + quantities[1:]
            yield final[0], final[1]
            yield final[2], final[3]
```
- we have a compute_quantities_for_vmap_test that calls `_compute_quantities_for_vmap_test` but does two yields instead of one.

Reply:
Updated.